evadb/functions/forecast.py

```diff
@@ -44,19 +46,20 @@ def setup(
         self.predict_column_rename = predict_column_rename
         self.time_column_rename = time_column_rename
         self.id_column_rename = id_column_rename
+        self.horizon = int(horizon)
+        self.library = library
 
     def forward(self, data) -> pd.DataFrame:
-        horizon = list(data.iloc[:, -1])[0]
-        assert (
-            type(horizon) is int
-        ), "Forecast UDF expects integral horizon in parameter."
-        forecast_df = self.model.predict(h=horizon)
+        if self.library == "statsforecast":
+            forecast_df = self.model.predict(h=self.horizon)
+        else:
+            forecast_df = self.model.predict()
         forecast_df.reset_index(inplace=True)
         forecast_df = forecast_df.rename(
             columns={
                 "unique_id": self.id_column_rename,
                 "ds": self.time_column_rename,
                 self.model_name: self.predict_column_rename,
             }
-        )
+        )[: self.horizon * forecast_df["unique_id"].nunique()]
```

Comment:
How does `neuralforecast` handle `unique_id` column? Is that similar to `statsforecast`?